pkg/toolsets/toolsets.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package toolsets
+
+import "strings"
+
+// Toolset represents a group of related tools
+type Toolset string
+
+const (
+	// Core toolsets
+	ToolsetRegistry        Toolset = "registry"         // Public Terraform Registry
+	ToolsetRegistryPrivate Toolset = "registry-private" // Private registry (TFE/TFC)
+	ToolsetTerraform       Toolset = "terraform"        // TFE/TFC operations
+
+	// Special toolsets
+	ToolsetAll     Toolset = "all"
+	ToolsetDefault Toolset = "default"
+)
+
+type ToolsetMetadata struct {
+	ID          string
```

Comment:
Above you defined a type for this, but here you're using string and calling it `ID`. If you rename you `Toolset` type to `ToolsetName` this line becomes 

```golang
Name ToolsetName
```